tasks/service_create.rb

```diff
@@ -6,18 +6,18 @@
 require 'puppet'
 
 def service_create(image, replicas, expose, env, command, extra_params, service, detach)
-  cmd_string = 'docker service create'
+  cmd_string = ['docker', 'service', 'create']
```

Comment:
We should try to avoid using += here.

I believe [Array#concat](https://ruby-doc.org/core-2.1.1/Array.html#method-i-concat) would be preferable in this instance.

This applies to all instances in this file.

Also lets rename the variable to cmd rather than cmd_string.

Reply:
See above.

Reply:
In this case you have changed from concatenating a string to an array so we should generally lean towards the more idiomatic approach.